mbed_greentea/mbed_greentea_cli.py

```diff
@@ -782,7 +782,9 @@ def get_parallel_value(value):
                 if not mbed_dev['serial_port'].endswith(str(baudrate)):
                     mbed_dev['serial_port'] = "%s:%d" % (mbed_dev['serial_port'], baudrate)
                 mut = mbed_dev
-                muts_to_test.append(mbed_dev)
+                if mbed_dev not in muts_to_test:
```

Comment:
The only concern here is if Python is able to check that the dictionary `mbed_dev` is equal to another dictionary in a list. I know it can do this for things like strings and numbers, but does it work for full dictionaries as well?